seaborn/axisgrid.py

```diff
@@ -461,13 +461,13 @@ def __init__(
         if despine:
             self.despine()
 
-        if sharex:
+        if sharex != 'row':
```

Comment:
Won’t this hide the labels when `sharex=False`?

Reply:
true. checked by eye again and fixed. I didnt find any tests for `sharex` and `sharey` in `axesgrid.py`. hope without tests this is ok.

```python
import matplotlib.pyplot as plt
import seaborn as sns
df = sns.load_dataset("titanic")
import matplotlib.pyplot as plt
for sharex in [True, False, 'row','col']:
    for sharey in [True, False, 'row','col']:
        g=sns.catplot(
            data=df, kind="point", sharex=sharex, sharey=sharey,
            x="class", y="fare", row='alive', col='sex')
        plt.suptitle(f'sharex={sharex} & sharey={sharey}')
        plt.show()
```